### PR TITLE
update files for winter 2023

### PR DESCRIPTION
**Pull request by eb-64-64**

### CODE REVIEW

File: key_file12.txt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+Checking that 1 is at the head of the tree...true
+Attempting to remove 1 from the tree...true
+
+Checking that 7 is at the head of the tree...true
+Attempting to remove 7 from the tree...true
+
+Pretty printing the tree...
+
+Empty tree
```
Comment (byubean) [inline key_file12.txt]: Let's have a separate test like this and remove this part (same for other similar tests). Most tests should focus on a single unit of functionality, with maybe 1 or 2 tests that focus on comprehensive performance.

File: tests.cpp [hunk 2]
```diff
@@ -0,0 +1,984 @@
...
+#include "AVL.h"
+#include "printing.h"
+
+void test1() {
+    std::cout << "--- Test 1 output ---" << std::endl;
+
+    AVL tree;
+
+    std::cout << "\nPretty printing the tree..." << std::endl;
+    pretty_print_tree(tree);
```
Comment (byubean) [inline tests.cpp]: You could include 
```c++
std::cout << std::endl << "Pretty printing the tree..." << std::endl;
```
in the definition of `pretty_print_tree`.

File: tests.cpp [hunk 3]
```diff
@@ -0,0 +1,984 @@
...
+
+    std::cout << "\nAttempting to insert 1 into the tree..." << std::boolalpha << tree.insert(1) << std::endl;
```
Comment (byubean) [inline tests.cpp]:
```c++
void insert(AVL &tree, int item) {
    std::cout << std::endl << "Inserting " << item << "..." << std::boolalpha << tree.insert(item) << std::endl;
}
```
And then use it all over. :)

Then do something similar for removing.

**Comment by eb-64-64** [comment]
I'm so sorry that these diffs are always so massive! 😬 These are the changes I made:
* Make the `contains` and `size` methods `const`
* Refactor the code in `tests.cpp` with helper functions for the operations done on the tree
* Clean up tests 3-10 (the `insert` tests that were more complicated)
* Get rid of the teardown code (that repeatedly removed the head) in the first 4 `remove` tests (tests 12-16). I left it in the more complicated `remove` test (test 17)
* Regenerate the key files

**Comment by eb-64-64** [comment]
I just realized something—I'm used to defining tree height as 0 if the node is null and 1 if the node is a leaf. But when I took this class, we defined the height as -1 if the node is null and 0 if the node is a leaf. The constructor I wrote for `Node` initializes `height` to 1, but should it instead initialize `height` to 0?

**Pull request merged**